From: Kristian Nielsen <knielsen@knielsen-hq.org>
Origin: https://gist.github.com/knielsen/6540045
Bug: http://bugs.mysql.com/bug.php?id=69458
Description: There is a bug in the test suite that leaves behind dirty
 temp files. This change forces mysqld to clean them up by disconnecting.

Index: mysql-5.5-5.5.33+dfsg/mysql-test/extra/rpl_tests/rpl_innodb.test
===================================================================
--- mysql-5.5-5.5.33+dfsg.orig/mysql-test/extra/rpl_tests/rpl_innodb.test	2013-07-15 05:01:49.000000000 -0700
+++ mysql-5.5-5.5.33+dfsg/mysql-test/extra/rpl_tests/rpl_innodb.test	2013-09-13 08:17:10.902875508 -0700
@@ -114,6 +114,11 @@
 FLUSH LOGS;
 
 DROP DATABASE mysqltest1;
+# The temporary table mysqltest1.tmp2 is only dropped on the slave when the
+# owning connection on the master is closed. So do that now to not leave a
+# temporary table file around after this test terminates.
+disconnect master;
+connect (master,127.0.0.1,root,,test,$MASTER_MYPORT,);
 
 --echo End of 5.1 tests
 
